server/tests/roadmaps.test.ts

```diff
@@ -2,89 +2,158 @@ import chai, { assert, expect } from 'chai';
+          rm.name == newRoadmap.name &&
+          rm.description == newRoadmap.description,
+      );
+
       expect(res.status).to.equal(200);
       expect(before.body.length + 1).to.equal(after.body.length);
-      const addedRoadmap = after.body.find(
-        (rm: any) => rm.name == 'TestRoadmap',
-      );
-      expect(addedRoadmap).to.exist;
+      expect(addedRoadmapApi).to.exist;
+      expect(addedRoadmapDb).to.exist;
     });
+    // Permission tests needed?
   });
 
   describe('PATCH /roadmaps/:roadmapId', function () {
     it('Should patch roadmap', async function () {
-      const firstRoadmapId = (await Roadmap.query().first()).id;
+      const roadmapToPatchId = (await Roadmap.query().first()).id;
+      const patchPayload = {
+        name: 'patchedName',
+        description: 'patchedDesc',
+      };
       const res = await loggedInAgent
-        .patch('/roadmaps/' + firstRoadmapId)
+        .patch('/roadmaps/' + roadmapToPatchId)
         .type('json')
-        .send({ name: 'patchedname', description: 'patcheddesc' });
-      const after = await loggedInAgent.get('/roadmaps/');
+        .send(patchPayload);
+
       expect(res.status).to.equal(200);
-      const patchedRoadmap = after.body.find(
-        (rm: any) => rm.name == 'patchedname',
+
+      // Check if database updated
+      const patchedRoadmapDb = await Roadmap.query().where(
+        'id',
+        roadmapToPatchId,
       );
-      expect(patchedRoadmap).to.exist;
-      expect(patchedRoadmap.description).to.equal('patcheddesc');
+
+      expect(patchedRoadmapDb[0].name).to.equal('patchedName');
+      expect(patchedRoadmapDb[0].description).to.equal('patchedDesc');
+
+      // Check if API updated
+      const after = await loggedInAgent.get('/roadmaps/');
+      const patchedRoadmapApi = after.body.find(
+        (rm: any) =>
+          rm.name == 'patchedName' &&
+          rm.description == 'patchedDesc' &&
+          rm.id == roadmapToPatchId,
+      );
+
+      expect(patchedRoadmapApi).to.exist;
     });
     it('Should not patch roadmap with incorrect permissions', async function () {
-      const firstRoadmapId = (await Roadmap.query().first()).id;
+      const firstRoadmap = await Roadmap.query().first();
+      const patchPayload = {
+        name: 'patchedName',
+        description: 'patchedDesc',
+      };
       const res = await withoutPermission(
-        firstRoadmapId,
+        firstRoadmap.id,
         Permission.RoadmapEdit,
         () =>
           loggedInAgent
-            .patch('/roadmaps/' + firstRoadmapId)
+            .patch('/roadmaps/' + firstRoadmap.id)
             .type('json')
-            .send({ name: 'patchedname', description: 'patcheddesc' }),
+            .send(patchPayload),
       );
       const after = await loggedInAgent.get('/roadmaps/');
-      expect(res.status).to.equal(403);
+      const firstRoadmapAfter = await Roadmap.query().where(
+        'id',
+        firstRoadmap.id,
+      );
```

Comment:
When selecting by id `findById` works nicely, it also returns max. 1 result
```suggestion
      const firstRoadmapAfter = await Roadmap.query().findById(firstRoadmap.id);
```

There was another one on line 110